### PR TITLE
Implement core NFT views to conform to Flow NFT Metadata standard

### PR DESCRIPTION
**Pull request by EricLin-wk**
Implement Core NFT Views from the MetadataViews contract  to conform to the Flow NFT Metadata standard https://www.flow-nft-catalog.com/

### CODE REVIEW

File: contracts/AllDay.cdc [hunk 1]
```diff
@@ -537,6 +549,196 @@ pub contract AllDay: NonFungibleToken {
...
+                    return MetadataViews.dictToTraits(dict: fullDictionary, excludedNames: excludedNames)
+            }
+            return nil
+        }
+
+        pub fun getName(): String {
+            let edition: EditionData = AllDay.getEditionData(id: self.editionID)
+            let play: PlayData = AllDay.getPlayData(id: edition.playID)
+            let firstName: String = play.metadata["PlayerFirstName"] ?? ""
+            let lasttName: String = play.metadata["PlayerLastName"] ?? ""
```
Comment (sadief) [inline contracts/AllDay.cdc:684]:
```suggestion
            let lastName: String = play.metadata["PlayerLastName"] ?? ""
```
tiny tiny nitpick

File: contracts/AllDay.cdc [hunk 2]
```diff
@@ -31,6 +33,16 @@ import NonFungibleToken from "./NonFungibleToken.cdc"
 // The AllDay NFTs and metadata contract
 //
 pub contract AllDay: NonFungibleToken {
+
+    //------------------------------------------------------------
+    // Deployment variables
+    //------------------------------------------------------------
+
+    // The address to which royalties should be deposited
+    pub fun RoyaltyAddress() : Address {
```
Comment (sadief) [inline contracts/AllDay.cdc:42]: Do we need to have a RoyaltyAddress in order to implement the metadata views?

File: lib/go/test/allday_test.go [hunk 3]
```diff
@@ -538,3 +550,116 @@ func testMintMomentNFT(
...
+
+		// Validate External URL
+		externalURLView := result[2]
+		assert.Equal(t, "https://nflallday.com/moments/1", externalURLView.Fields[0].ToGoValue())
+
+		//Validate Medias
+		mediasView := result[3]
+		editions := mediasView.Fields[0].(cadence.Array)
+		assert.Equal(t, "https://assets.nflallday.com/editions/1/media/image?format=jpeg&width=512",
+			editions.Values[0].(cadence.Struct).Fields[0].(cadence.Struct).Fields[0].ToGoValue())
```
Comment (sadief) [inline lib/go/test/allday_test.go:590]: nice tests! You could have a helper func that takes in the index and `editions` and returns the result of 
```
editions.Values[0].(cadence.Struct).Fields[0].(cadence.Struct).Fields[0].ToGoValue())
```

that you could use in the assert.Equal just to make the tests a little less verbose

File: contracts/AllDay.cdc (hunk 2 above)
Comment (EricLin-wk) [reply contracts/AllDay.cdc:42]: No, I'll get rid of this method and move the variable into `resolveView()`

**Review by sadief — APPROVED** [review]
Nice! Non blocking, but would you mind adding some info to the Readme about the metadata views as well?

**Pull request merged**